Resources/public/js/views/ez-subitemlistview.js

```diff
@@ -160,6 +173,164 @@ YUI.add('ez-subitemlistview', function (Y) {
+        _displayPriorityButtons: function (e) {
+            var selectedCell = this.get('container').one('#priority-cell-' + e.target.getAttribute('data-location-id'));
+
+            e.preventDefault();
+            if (!this.get('isOnePrioritySelected')) {
+                this._set('isOnePrioritySelected', true);
+                selectedCell.removeClass('ez-subitem-hovered-priority-cell');
+                selectedCell.addClass('ez-subitem-selected-priority-cell');
+                e.target.removeAttribute('readonly');
+            }
+        },
+
+        /**
+         * Hide the 'validate' & 'cancel' buttons which allow to update or not the priority.
+         * Also fill back priority input with the current sub item priority.
+         *
+         * @method _hidePriorityButtons
+         * @protected
+         * @param {Object} e event facade
+         */
+        _hidePriorityButtons: function (e) {
+            var selectedCell = this.get('container').one('.ez-subitem-selected-priority-cell'),
+                input = this.get('container').one('#' + e.target.getAttribute('data-priority-input'));
+
+            Y.Array.each(this.get('subitems'), Y.bind(function (locStruct) {
+                if (locStruct.location.get('locationId') == input.getAttribute('data-location-id')) {
+                    input.set('value', locStruct.location.get('priority'));
+                    input.setAttribute('readonly', 'readonly');
+                }
+            },this));
+
+            e.preventDefault();
+            selectedCell.removeClass('ez-subitem-selected-priority-cell');
+            this._set('isOnePrioritySelected', false);
+        },
+
+        /**
+         * Event handler allowing to change the priority of the subitem
+         *
+         * @method _setPriority
+         * @protected
+         * @param {Object} e event facade
+         */
+        _setPriority: function (e) {
+            var button = e.target,
+                selectedCell = this.get('container').one('.ez-subitem-selected-priority-cell'),
+                input;
+
+            e.preventDefault();
+            input = this.get('container').one('#' + button.getAttribute('data-priority-input'));
+
+            Y.Array.each(this.get('subitems'), Y.bind(function (locStruct) {
+                if (locStruct.location.get('locationId') == input.getAttribute('data-location-id')) {
+                    //locStruct.location.updatePriority();
+                    this.fire('updatePriority', {item: locStruct, priority: input.get('value')});
+                }
+            },this));
+            input.setAttribute('readonly', 'readonly');
+            selectedCell.removeClass('ez-subitem-selected-priority-cell');
+            this._set('isOnePrioritySelected', false);
+        },
+
+        /**
+         * Validates the current input of priority
+         *
+         * @method validate
+         * @param {Object} e event facade
+         */
+        validate: function (e) {
```

Comment:
why is  this method public ? also its name is very imprecise, what does it validate ?